pkg/skaffold/build/tag/tag.go

```diff
@@ -32,21 +35,39 @@ type Tagger interface {
 	GenerateTag(workingDir, imageName string) (string, error)
 }
 
+const DeprecatedImageName = "_DEPRECATED_IMAGE_NAME_"
+
 // GenerateFullyQualifiedImageName resolves the fully qualified image name for an artifact.
 // The workingDir is the root directory of the artifact with respect to the Skaffold root,
 // and imageName is the base name of the image.
 func GenerateFullyQualifiedImageName(t Tagger, workingDir, imageName string) (string, error) {
+	// Supporting the use of the deprecated {{.IMAGE_NAME}} in envTemplate
+	if v, ok := t.(*envTemplateTagger); ok {
+		tag, err := v.GenerateTag(workingDir, DeprecatedImageName)
+
+		if err != nil {
+			return "", fmt.Errorf("generating deprecated envTemplate tag: %w", err)
+		}
+
+		if strings.Contains(tag, DeprecatedImageName) {
+			warnings.Printf("{{.IMAGE_NAME}} is deprecated, envTemplate's template should only specify the tag value. See https://skaffold.dev/docs/pipeline-stages/taggers/")
+
+			tag, _ = v.GenerateTag(workingDir, imageName)
```

Comment:
This call to `GenerateTag` could still return an error and so we should handle it.

Reply:
Okay. Could you give me an example of when this would error if the first call to the function didn't error? Since we are only changing the value associated with `imageName`.

Reply:
It may not return an error *right now* but the implementation could change in the future.  We should strive to avoid depending on knowledge of function implementations.

Reply:
A safer alternative would be to do a string replacement on the tag to replace the `_DEPRECATED_IMAGE_NAME_`.

Reply:
I agree. This seems like a more optimal approach.